packages/firestore/src/local/shared_client_state.ts

```diff
@@ -423,8 +571,81 @@ export class WebStorageSharedClientState implements SharedClientState {
    * Parses a client state key in LocalStorage. Returns null if the key does not
    * match the expected key format.
    */
-  private fromLocalStorageClientKey(key: string): string | null {
+  private fromLocalStorageClientStateKey(key: string): ClientKey | null {
     const match = this.clientStateKeyRe.exec(key);
     return match ? match[1] : null;
   }
+
+  /**
+   * Parses a client state in LocalStorage. Returns null if the key does not
+   * match the expected key format.
```

Comment:
Per my comment above, I'm not a fan of these methods. If we keep them I'd suggest we put `try` in the method name to make it clearer that they're expected to be called with "incorrect" values and will just return null.